examples/C/color/rs-color.c

```diff
@@ -0,0 +1,127 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2017 Intel Corporation. All Rights Reserved.
+
+/* Include the librealsense C header files */
+#include <librealsense2/rs.h>
+#include <librealsense2/h/rs_pipeline.h>
+#include <librealsense2/h/rs_frame.h>
+
+#include <stdlib.h>
+#include <stdint.h>
+#include <stdio.h>
+#include "../example.h"
+
+int main()
+{
+    rs2_error* e = 0;
+    /* Create a context object. This object owns the handles to all connected realsense devices. */
+    rs2_context* ctx = rs2_create_context(RS2_API_VERSION, &e);
+    check_error(e);
+
+    /* Get a list of all the connected devices. */
+    rs2_device_list* device_list = rs2_query_devices(ctx, &e);
+    check_error(e);
+
+    int dev_count = rs2_get_device_count(device_list, &e);
+    check_error(e);
+    printf("There are %d connected RealSense devices.\n", dev_count);
+    if (0 == dev_count)
+        return EXIT_FAILURE;
+
+    rs2_device* dev = rs2_create_device(device_list, 0, &e);
+    check_error(e);
+
+    print_device_info(dev);
+
+    const rs2_stream stream = RS2_STREAM_COLOR;
+    const rs2_format format = RS2_FORMAT_RGB8;
+    const int width = 640;
+    const int height = 480;
+    const int fps = 30;
+    const int index = 0;
+
+    // Create a pipeline to configure, start and stop camera streaming
+    rs2_pipeline* pipeline =  rs2_create_pipeline(ctx, &e);
+    check_error(e);
+
+    // Create a config instance
+    rs2_config* config = rs2_create_config(&e);
+    check_error(e);
+
+    // Request a specific configuration
+    rs2_config_enable_stream(config, stream, index, width, height, format, fps, &e);
+    check_error(e);
+
+    // Check if current device supports color streaming
+    int supports_color = rs2_config_can_resolve(config, pipeline, &e);
```

Comment:
Somewhat redundant - the pipeline will return error on start if not supported and it will be handled anyway